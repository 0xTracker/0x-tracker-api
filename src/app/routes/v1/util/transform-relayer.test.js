const starBit = require('../../../../fixtures/relayers/star-bit');
const transformRelayer = require('./transform-relayer');

const stats = {
  tradeVolume: 150567.87,
  tradeCount: 512,
  fillVolume: 18907,
  fillCount: 687,
};

describe('transformRelayer', () => {
  it('should transfer relayer with stats', () => {
    const transformed = transformRelayer(starBit, stats);

    expect(transformed).toMatchSnapshot();
  });

  it('should transform relayer without stats', () => {
    const transformed = transformRelayer(starBit);

    expect(transformed).toMatchSnapshot();
  });

  it('should transform relayer with minimal fields', () => {
    const transformed = transformRelayer(
      { ...starBit, imageUrl: undefined, url: undefined },
      stats,
    );

    expect(transformed).toMatchSnapshot();
  });
});
