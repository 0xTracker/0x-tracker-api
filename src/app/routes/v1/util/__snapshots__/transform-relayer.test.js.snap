// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformRelayer should transfer relayer with stats 1`] = `
Object {
  "id": "starBit",
  "imageUrl": "https://resources.0xtracker.com/logos/starbit.png",
  "name": "STAR BIT",
  "slug": "star-bit",
  "stats": Object {
    "tradeCount": 512,
    "tradeVolume": 150567.87,
  },
  "url": "https://www.starbitex.com",
}
`;

exports[`transformRelayer should transform relayer with minimal fields 1`] = `
Object {
  "id": "starBit",
  "imageUrl": null,
  "name": "STAR BIT",
  "slug": "star-bit",
  "stats": Object {
    "tradeCount": 512,
    "tradeVolume": 150567.87,
  },
  "url": null,
}
`;

exports[`transformRelayer should transform relayer without stats 1`] = `
Object {
  "id": "starBit",
  "imageUrl": "https://resources.0xtracker.com/logos/starbit.png",
  "name": "STAR BIT",
  "slug": "star-bit",
  "url": "https://www.starbitex.com",
}
`;
